client/dom/radiobutton.ts

```diff
@@ -0,0 +1,122 @@
+import { Elem, Input } from '../types/d3'
+
+/* makes radio buttons */
+
+type RadioButtonOpts = {
+	/** Fires .oninput. Intended for general use. */
+	callback?: (f?: any) => void
+	/** Required.  */
+	holder: any
+	/** common Name of <input>, use random number if not given */
+	inputName?: string | number
+	/** Mass ui specific logic. Optional callback methods for
+	 * non oninput events. Intended to address needs for assistive techology
+	 */
+	listeners?: {
+		/** Fires on onmouseup and onkeyup for the button and text label
+		 */
+		input: () => void
+	}
+	/** arr of objs defining the radio buttons and properties */
+	options: OptionEntry[]
+	/** css to be applied to each <div> of the options
+	 * e.g. { "padding":"5px", "display":"inline-block" }
+	 */
+	styles?: { [index: string]: string | number }
+}
+
+type OptionEntry = {
+	/** only set for only *one* option */
+	checked?: boolean | number
+	/** Text shown in the span to the right of the radio button */
+	label: string
+	/** Text shown in tooltip */
+	title?: string
+	/** Should correspond to 'currValue' in callbacks */
+	value: string | number
+}
+
+type ReturnedRadios = {
```

Comment:
@creilly8 Nitpicking a bit, but a better type name will be `RadioApi`, since this is the "api" that a consumer would code against.